xbmc/pvr/channels/PVRChannelGroup.cpp

```diff
@@ -469,11 +475,71 @@ int CPVRChannelGroup::LoadFromDb()
   if (!database)
     return -1;
 
-  int iChannelCount = Size();
+  const int iChannelCount = Size();
+  const std::vector<std::shared_ptr<CPVRChannelGroupMember>> results = database->Get(*this);
+
+  std::vector<std::shared_ptr<CPVRChannelGroupMember>> membersToDelete;
+  if (!results.empty())
+  {
+    CSingleLock lock(m_critSection);
+    for (const auto& member : results)
+    {
+      // Consistency check.
+      if (member->ClientID() > 0 && member->ChannelUID() > 0 && member->IsRadio() == IsRadio())
+      {
+        m_sortedMembers.emplace_back(member);
+        m_members.emplace(
+            std::make_pair(std::make_pair(member->ClientID(), member->ChannelUID()), member));
+      }
+      else
+      {
+        CLog::LogF(LOGWARNING,
+                   "Skipping member with channel database id {} of {} channel group '{}'. "
+                   "Channel not found in the database or radio flag changed.",
+                   member->ChannelDatabaseID(), IsRadio() ? "radio" : "TV", GroupName());
+        membersToDelete.emplace_back(member);
+      }
+    }
+
+    SortByChannelNumber();
+  }
+
+  DeleteGroupMembersFromDb(membersToDelete);
+
+  return results.size() - membersToDelete.size() - iChannelCount;
+}
+
+void CPVRChannelGroup::DeleteGroupMembersFromDb(
+    const std::vector<std::shared_ptr<CPVRChannelGroupMember>>& membersToDelete)
+{
+  if (!membersToDelete.empty())
+  {
+    const std::shared_ptr<CPVRDatabase> database = CServiceBroker::GetPVRManager().GetTVDatabase();
+    if (!database)
+    {
+      CLog::LogF(LOGERROR, "No TV database");
+      return;
+    }
+
+    // Note: We must lock the db the whole time, otherwise races may occur.
+    database->Lock();
```

Comment:
Such naked Lock/Unlock always give me a shiver... What are the thoughts on renaming these member functions to lowercase `lock` and `unlock` so that a `std::lock_guard` can be used?

Reply:
This is something for another day/PR.